app/database/permission_granted_store_compute_all_access.go

```diff
@@ -0,0 +1,146 @@
+		INSERT INTO permissions_propagate (group_id, item_id, propagate_access)
+		SELECT
+			parents.group_id,
+			items_items.child_item_id,
+			'self' as propagate_access
+		FROM items_items
+		JOIN permissions_generated AS parents
+			ON parents.item_id = items_items.parent_item_id
+		JOIN permissions_propagate AS parents_propagate
+			ON parents_propagate.group_id = parents.group_id AND parents_propagate.item_id = parents.item_id AND
+				 parents_propagate.propagate_access = 'children'
+		ON DUPLICATE KEY UPDATE propagate_access='self'`
+	stmtMarkChildrenOfChildrenAsSelf, err = s.db.CommonDB().Prepare(queryMarkChildrenOfChildrenAsSelf)
+	mustNotBeError(err)
+	defer func() { mustNotBeError(stmtMarkChildrenOfChildrenAsSelf.Close()) }()
+
+	// deleting 'children' groups_items_propagate
+	const queryDeleteProcessedChildren = `DELETE FROM permissions_propagate WHERE propagate_access = 'children'`
+	stmtDeleteProcessedChildren, err = s.db.CommonDB().Prepare(queryDeleteProcessedChildren)
+	mustNotBeError(err)
+	defer func() { mustNotBeError(stmtDeleteProcessedChildren.Close()) }()
+
+	// computation for group-item pairs marked as 'self' in permissions_propagate (so normally all of them)
+	const queryUpdatePermissionsGenerated = `
+		INSERT INTO permissions_generated
+			(group_id, item_id, can_view_generated, can_grant_view_generated, can_watch_generated, can_edit_generated, is_owner_generated)
+		SELECT
+			permissions_propagate.group_id,
+			permissions_propagate.item_id,
+			IF(granted.is_owner, 'solution', GREATEST(
+				IFNULL(granted.can_view_value, 1),
+				IFNULL(new_data.can_view_value, 1)
+			)) AS can_view_generated,
+			IF(granted.is_owner, 'transfer', GREATEST(
+				IFNULL(granted.can_grant_view_value, 1),
+				IFNULL(new_data.can_grant_view_value, 1)
+			)) AS can_grant_view_generated,
+			IF(granted.is_owner, 'transfer', GREATEST(
+				IFNULL(granted.can_watch_value, 1),
+				IFNULL(new_data.can_watch_value, 1)
+			)) AS can_watch_generated,
+			IF(granted.is_owner, 'transfer', GREATEST(
+				IFNULL(granted.can_edit_value, 1),
+				IFNULL(new_data.can_edit_value, 1)
+			)) AS can_edit_generated,
+			IFNULL(granted.is_owner, 0) AS is_owner_generated
+		FROM permissions_propagate
+		LEFT JOIN LATERAL (
+			SELECT
+				MAX(LEAST(
+					IF(LEAST(items_items.descendants_and_solution_view_propagation_value+2, parent.can_view_generated_value) = 3,
+						items_items.content_view_propagation_value,
+						IF(parent.can_view_generated = 'info', 1, parent.can_view_generated_value)),
+					items_items.descendants_and_solution_view_propagation_value+2)) AS can_view_value,
+				MAX(IF(items_items.grant_view_propagation, LEAST(parent.can_grant_view_generated_value, 4), 1)) AS can_grant_view_value,
+				MAX(IF(items_items.watch_propagation, LEAST(parent.can_watch_generated_value, 3), 1)) AS can_watch_value,
+				MAX(IF(items_items.edit_propagation, LEAST(parent.can_edit_generated_value, 3), 1)) AS can_edit_value
```

Comment:
For these 3, probably it would be nice to put the enum value as inline comment, such as 
```sql
MAX(IF(items_items.grant_view_propagation, LEAST(parent.can_grant_view_generated_value, 4 /* solution */), 1)) AS can_grant_view_value,
```

Reply:
done